doc/source/whatsnew/v2.1.0.rst

```diff
@@ -225,6 +225,8 @@ Styler
 
 Other
 ^^^^^
+- Bug in :meth:`DataFrame.resample` and :meth:`Series.resample` in incorrectly allowing non-fixed ``freq`` when resampling on a :class:`TimedeltaIndex` (:issue:`51896`)
```

Comment:
Should probably go in `Groupby/resample/rolling`

Reply:
moved+green